### DISCUSSION TITLE
Using different type of filed for same attribute field, in different situation, Confilcts

### DISCUSSION THREAD

**Discussion by ehs4nm** [filamentphp/filament#1793] [Help] [answered]
I know I'm doing something wrong but i need to provide two different way to select a parameter like client_id, one for admin to select and one as a hidden field to catch it from auth() for user. 
```
                Select::make('client_id')->options(User::all()->pluck('id','name')->hidden(auth()->user()->hasRole('super_admin')),
                Hidden::make('client_id')->default(fn () => auth()->user()->profileable->id ?? null),
```
No surprise always the last field override the previous. but how can i do this? (for admin show select filed and for user nothing, just a hidden parameter)

**Comment by danswiser** [comment]
From my understanding of your issue here's my thoughts:

1. [Why its not working] You can only submit one input by name in an HTTP request. If there are multiple values using the same name key ('client_id') then the last one is what's used. It looks like the hidden field always exists, and is the last in the order, so it will always be the value that gets submitted with the form.
2. [Swap field order] If you put the hidden field before the select input then the selected value will override the hidden value when the select input is visible and an option is selected.
3. [Keep field order] If you hide the select field, then the hidden field will exist. If the select field is visible then the hidden field should not exist.

**Comment by danharrin** [comment, marked as answer]
Tried just combining them?

```php
Select::make('client_id')
    ->options(User::all()->pluck('id', 'name'))
    ->hidden(auth()->user()->hasRole('super_admin')),
    ->default(fn () => auth()->user()->profileable->id ?? null),
```

**Reply by ehs4nm** [reply]
It is a better and simpler approach. thanks.
